os/net/lwip/src/core/ipv4/dhcp.c

```diff
@@ -694,6 +702,28 @@ void dhcp_cleanup(struct netif *netif)
 	}
 }
 
+#if LWIP_NETIF_HOSTNAME
+void dhcp_hostname(struct netif *netif, char *name)
+{
+	char *name_heap;
+
+	if (netif == NULL || name == NULL) {
+		return;
+	}
+
+	if (netif->hostname != NULL) {
+		kmm_free(netif->hostname);
+		netif->hostname = NULL;
+	}
+
+	name_heap = (char *)kmm_malloc(strlen(name) + 1);
+	memcpy(name_heap, name, strlen(name));
```

Comment:
how about strcpy instead of memcpy & null adding?

Reply:
Applied. Thanks for the command.
But I put NULL at the last, for safety.